server/server.ts

```diff
@@ -1,8 +1,30 @@
 /* eslint-disable @typescript-eslint/no-unused-vars -- Remove when used */
 import 'dotenv/config';
-import express from 'express';
-import pg from 'pg';
-import { ClientError, errorMiddleware } from './lib/index.js';
+import express, { response } from 'express';
+import pg, { Client } from 'pg';
+import { ClientError, errorMiddleware, authMiddleware } from './lib/index.js';
+import OpenAI from 'openai';
+import { z } from 'zod';
+import argon2, { hash } from 'argon2';
+import jwt, { TokenExpiredError } from 'jsonwebtoken';
+
+const openai = new OpenAI({
+  apiKey: process.env.OPENAI_API_KEY,
+});
+type User = {
+  userId: number;
+  username: string;
+  hashedPassword: string;
+};
+type Auth = {
+  username: string;
+  password: string;
+};
+
+const hashKey = process.env.TOKEN_SECRET;
+if (!hashKey) throw new Error('TOKEN_SECRET not found in .env');
+
+const googleKey = process.env.REACT_APP_GOOGLE_MAPS_API_KEY;
```

Comment:
You might want to use the same technique that is used in hashKey to make sure it exists in the environment variables. (That is, repeat line 25 for googleKey)